tests/test_solver.py

```diff
@@ -54,28 +54,66 @@ def test_ScipyLBFGSB_with_gradient():
     sol_ref = np.array([1.0, 1.0, 1.0, 1.0, 1.0])
     assert np.allclose(sol, sol_ref)
 
-def test_ScipyLeastSquares_without_Jac():
+def test_ScipyLSQ_without_Jac():
     def fun_rosenbrock(x):
         return np.array([10 * (x[1] - x[0]**2), (1 - x[0])])
     x0 = np.array([2, 2])
-    solver = ScipyLeastSquares(fun_rosenbrock, x0)
+    solver = ScipyLSQ(fun_rosenbrock, x0)
     sol, _ = solver.solve()
     sol_ref = np.array([1, 1])
     assert np.allclose(sol, sol_ref)
 
-def test_ScipyLeastSquares_with_Jac():
+def test_ScipyLSQ_with_Jac():
     def fun_rosenbrock(x):
         return np.array([10 * (x[1] - x[0]**2), (1 - x[0])])
     def jac_rosenbrock(x):
         return np.array([
             [-20 * x[0], 10],
             [-1, 0]])
     x0 = np.array([2, 2])
-    solver = ScipyLeastSquares(fun_rosenbrock, x0, jacfun=jac_rosenbrock)
+    solver = ScipyLSQ(fun_rosenbrock, x0, jacfun=jac_rosenbrock)
     sol, _ = solver.solve()
     sol_ref = np.array([1, 1])
     assert np.allclose(sol, sol_ref)
 
+def test_ScipyLinearLSQ_with_matrix():
+    rng = np.random.default_rng(seed = 1219)
+    m, n = 10, 5
+    A = rng.standard_normal((m, n))
+    b = rng.standard_normal(m)
+    res = lsq_linear(A, b, tol=1e-8)
+    ref_sol = res.x
+    sol, _ = ScipyLinearLSQ(A, b).solve()
+    assert np.allclose(sol, ref_sol, rtol=1e-3)
```

Comment:
just curious, in this test and the next, can we tighten the rtol to for example 1e-10, since it is the same underlying algorithm? (might need to fix the seed before each to use the same seed before each solve)